core/cmd/remote_client_test.go

```diff
@@ -188,6 +188,46 @@ func TestClient_CreateServiceAgreement(t *testing.T) {
 	}
 }
 
+func TestClient_CreateExternalInitiator(t *testing.T) {
+	t.Parallel()
+
+	app, cleanup := cltest.NewApplicationWithKey(t)
+	defer cleanup()
+	client, _ := app.NewClientAndRenderer()
+
+	tests := []struct {
+		name    string
+		args    []string
+		errored bool
+	}{
+		{"no arguments", []string{}, true},
+		{"not enough arguments", []string{"bitcoin"}, true},
+		{"too many arguments", []string{"bitcoin", "https://valid.url", "extra arg"}, true},
+		{"invalid url", []string{"bitcoin", "not a url"}, true},
+		{"valid external initiator creation", []string{"bitcoin", "http://testing.com/external_initiators"}, false},
+	}
+
+	for _, tt := range tests {
+		test := tt
+		t.Run(test.name, func(t *testing.T) {
+
+			set := flag.NewFlagSet("create", 0)
+			assert.NoError(t, set.Parse(test.args))
+			c := cli.NewContext(nil, set, nil)
+
+			err := client.CreateExternalInitiator(c)
+
+			cltest.AssertError(t, test.errored, err)
+			exis := cltest.AllExternalInitiators(t, app.Store)
+			if !test.errored {
```

Comment:
This is IMHO a bad pattern we've been trying to move away from. I think a better pattern is to split your tables for tests with errors / vs success states. e.g:

```
func TestClient_CreateExternalInitiator(t *testing.T) {
  ...
  assert.NoError()
...
func TestClient_CreateExternalInitiator_Errors(t *testing.T) {
...
  assert.Error(
```

If I see logic in a test I usually take that as a sign to refactor the test.

Reply:
Sure, what's the rationale behind this?

Reply:
Logic should be tested, writing tests for tests is hard, so we should try to avoid logic in tests. It makes tests slightly harder to read since you have to reason about which path it's taking.